### ISSUE TITLE
Script for Get List of Stale Devices and Remove them

### ISSUE THREAD

**Issue by subbzz** [MrTechGadget/aw-bulkdevices-script#7]
Hi MrTechGadget,

As discussed via mail. I am wondering if we can have a new script to identify and remove stale entries (from all OGs) beyond x number of days (say devices that did not show up over 25 days). This script can run on a daily basis. The date and time can be pegged from the server this script is running. 

Thanks,
SS

**Comment by MrTechGadget** [comment]
@subbzz , thanks for opening this. I'm thinking about this and would like some additional info. 

1. Does it matter the status of the device? (Enrolled, Unenrolled, Pending Enterprise Wipe, Device Wipe Initiated, etc.) 
2. Do you want to wipe enrolled devices first or just delete the record?
2b. If yes, would it be different for employee vs company owned, and different for DEP supervised devices?
3. What device details are important to log before deletion?

**Comment by subbzz** [comment]
Hi Mr TechGadget,

Many thanks for your response.

Basically I modified your code and now I am able to get it working the way
I wanted.

I completed the testing only last evening.

Many thanks again.

Cheers
Siva

**Comment by subbzz** [comment]
Sure. I will add it to your collection over the weekend.

Just wanted to check with you something for another use case, I wanted to
fetch user id for an user, and I am using username but the api is returning
not the exact match with user name. I am getting results all values match
multiple results. For example, User1 (Andrew.jackson@email.com / ajack) and
User 2 (Alex.jackson@email.com / ajack1), api query with username is
returning both results.

Any ideas to get the results based on exact match of username?

Cheers,
Siva

**Comment by MrTechGadget** [comment]
Yes, username query returns partial matches and this field is not unique.
You have to perform string comparisons (be sure to remember case) on the
results to grab the exact match.
